AdventOfCode2022/Program.cs

```diff
@@ -11,5 +66,120 @@ static void Main()
+                    }
+                    break;
+
+                case 'D':
+                    for (int j = 0; j < distance; j++)
+                    {
+                        coorH[1]--;
+                        coorT1 = p.Check(coorH, coorT1);
+                        coorT2 = p.Check(coorT1, coorT2);
+                        coorT3 = p.Check(coorT2, coorT3);
+                        coorT4 = p.Check(coorT3, coorT4);
+                        coorT5 = p.Check(coorT4, coorT5);
+                        coorT6 = p.Check(coorT5, coorT6);
+                        coorT7 = p.Check(coorT6, coorT7);
+                        coorT8 = p.Check(coorT7, coorT8);
+                        coorT9 = p.Check(coorT8, coorT9);
+
+                        List<int> copyCoorT1 = new List<int>(coorT1);
+                        List<int> copyCoorT9 = new List<int>(coorT9);
+                        coorSave1.Add(copyCoorT1);
+                        coorSave9.Add(copyCoorT9);
+                    }
+                    break;
+                case 'R':
+                    for (int j = 0; j < distance; j++)
+                    {
+                        coorH[0]++;
+                        coorT1 = p.Check(coorH, coorT1);
+                        coorT2 = p.Check(coorT1, coorT2);
+                        coorT3 = p.Check(coorT2, coorT3);
+                        coorT4 = p.Check(coorT3, coorT4);
+                        coorT5 = p.Check(coorT4, coorT5);
+                        coorT6 = p.Check(coorT5, coorT6);
+                        coorT7 = p.Check(coorT6, coorT7);
+                        coorT8 = p.Check(coorT7, coorT8);
+                        coorT9 = p.Check(coorT8, coorT9);
+
+                        List<int> copyCoorT1 = new List<int>(coorT1);
+                        List<int> copyCoorT9 = new List<int>(coorT9);
+                        coorSave1.Add(copyCoorT1);
+                        coorSave9.Add(copyCoorT9);
+                    }
+                    break;
+                case 'L':
+                    for (int j = 0; j < distance; j++)
+                    {
+                        coorH[0]--;
+                        coorT1 = p.Check(coorH, coorT1);
+                        coorT2 = p.Check(coorT1, coorT2);
+                        coorT3 = p.Check(coorT2, coorT3);
+                        coorT4 = p.Check(coorT3, coorT4);
+                        coorT5 = p.Check(coorT4, coorT5);
+                        coorT6 = p.Check(coorT5, coorT6);
+                        coorT7 = p.Check(coorT6, coorT7);
+                        coorT8 = p.Check(coorT7, coorT8);
+                        coorT9 = p.Check(coorT8, coorT9);
+
+                        List<int> copyCoorT1 = new List<int>(coorT1);
+                        List<int> copyCoorT9 = new List<int>(coorT9);
+                        coorSave1.Add(copyCoorT1);
+                        coorSave9.Add(copyCoorT9);
+                    }
+                    break;
+            }
```

Comment:
You could reduce this switch to only determine index of coorH and +1/-1 operation and reduce for loops to be implemented once after the switch